File: PetitionGenerator.java
import java.util.Set;
import java.io.FileInputStream;
import java.io.FileNotFoundException;

public class PetitionGenerator {
  private Set<String> lastNames;
  private Set<String> firstName;
  static private String defaultFile = "rawname.txt";
  // set.toArray()[rand.nextInt(set.size())]

  public PetitionGenerator() {
    this(defaultFile);
  }

  public PetitionGenerator(String fileName) {
    // Open the file
    try {
      FileInputStream fstream = new FileInputStream(fileName);
    } catch (FileNotFoundException e) {
      System.out.println("Names file not found, plz check if the file exist.");
    }
  }

  public static void main(String[] args) {
    System.out.println("Helo");
    PetitionGenerator pg = new PetitionGenerator();
  }
}

File: PetitionGenerator.scala
import scala.io.Source
object PetitionGenerator{
    def usage() {
        Console.err.println("Usage: PetitionGenerator rawname.txt count")
    }

    def main(args: Array[String]) {
        if (args.length != 2) {
            usage()
            sys.exit(1)
        }
        val random = new scala.util.Random(System.nanoTime)
        val lastNames = scala.collection.mutable.Set[Char]()
        val firstNames = scala.collection.mutable.Set[Char]()
        val fileName = args(0)
        val count = args(1).toInt
        for (fullName <- Source.fromFile(fileName).getLines()) {
            val lastName = fullName(0)
            val firstName = fullName.drop(1)
            for (c <- firstName) firstNames += c
            lastNames += lastName
        }
        val firstNamesList = firstNames.toList
        val lastNamesList = lastNames.toList

        1 to count foreach {
            _ => 
            val fakeName = new StringBuilder
            fakeName += lastNamesList(random.nextInt(lastNamesList.size))
            fakeName += firstNamesList(random.nextInt(firstNamesList.size))
            fakeName += firstNamesList(random.nextInt(firstNamesList.size))
            println(s"$fakeName")
        }
    }
}
